### PR TITLE
Add a threshold setting for number of episodes already downloaded to …

### PR DESCRIPTION
**Pull request by jawilson**
Adds a threshold to `next_series_seasons` to specify how many episodes may be downloaded for a particular season before emitting a season pack.

### Motivation for changes:

If several episodes of a season are already downloaded, I don't usually want to download the entire season pack.

### Detailed changes:
- Added `threshold` property to `next_series_seasons`
- If `series.episodes_for_season()` is greater than the threshold, skip the season

### Config usage if relevant (new plugin or updated schema):
```
tasks:
  discover-tv:
    discover:
      release_estimations:
        optimistic: 30 days
      what:
        - next_series_seasons:
            backfill: yes
            from_start: yes
            threshold: 4
      from:
        - piratebay:
            category: "highres tv"
            sort_by: seeds
      interval: 1 day
```

### CODE REVIEW

File: flexget/plugins/input/next_series_seasons.py [hunk 1]
```diff
@@ -81,6 +82,8 @@ def on_task_input(self, task, config):
         else:
             self.rerun_entries = []
 
+        threshold = config.get('threshold', None)
```
Comment (cvium) [inline flexget/plugins/input/next_series_seasons.py]: `None` is already the default value if key doesn't exist.

**Comment by liiight** [comment]
You can already set threshold in series season pack config. It is set to 0 by default which means that if any episodes of a season were downloaded, season packs will be rejected.

How is this different?

**Comment by jawilson** [comment]
@liiight the difference is that this configures the threshold for _emitting_ season from `next_series_seasons`, the `series` threshold only configures the threshold for _accepting_. This is the same situation for `backfill` and `from_start`.

The reason for this change is that I don't want to emit seasons for searching (causing extra calls to my search providers) for seasons that I would never end up accepting.

The other option would be to use the `series` config to control the `next_series_seasons` and `next_series_episodes`, which does make sense to me, but would be a change to existing config properties.

**Comment by liiight** [comment]
Bah, having the same config setup in more than one plugin sucks, but I guess it's an edge case.

**Comment by liiight** [comment]
Cool. Add to wiki?

**Comment by jawilson** [comment]
Finally got around to adding it to the wiki.

**Pull request merged**